src/main/cpp/auton/DriveToSpeaker.cpp

```diff
@@ -0,0 +1,94 @@
+//====================================================================================================================================================
+// Copyright 2024 Lake Orion Robotics FIRST Team 302
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+// OR OTHER DEALINGS IN THE SOFTWARE.
+//=====================================================================================================================================================
+
+// Team302 includes
+#include "auton/DriveToSpeaker.h"
+#include "chassis/ChassisConfigMgr.h"
+#include "chassis/DragonDriveTargetFinder.h"
+#include "pathplanner/lib/path/PathPlannerTrajectory.h"
+#include "chassis/driveStates/RobotDrive.h"
+#include "chassis/ChassisMovement.h"
+
+#include "utils/FMSData.h"
+
+// third party includes
+#include "pathplanner/lib/path/PathPlannerPath.h"
+
+DriveToSpeaker::DriveToSpeaker(RobotDrive *robotDrive,
+                               TrajectoryDrivePathPlanner *trajectoryDrivePathPlanner) : TrajectoryDrivePathPlanner(robotDrive),
+                                                                                         m_chassis(nullptr)
+{
+    auto config = ChassisConfigMgr::GetInstance()->GetCurrentConfig();
+    m_chassis = config != nullptr ? config->GetSwerveChassis() : nullptr;
+    m_dragonDriveTargetFinder = DragonDriveTargetFinder::GetInstance();
+    m_trajectoryDrivePathPlanner = trajectoryDrivePathPlanner;
+}
+
+void DriveToSpeaker::Init(ChassisMovement &chassisMovement)
+{
+    if (m_chassis != nullptr)
+    {
+        auto currentPose2d = m_chassis->GetPose();
+        auto aprilTagInfo = m_dragonDriveTargetFinder->GetPose(DragonVision::AMP);
+        auto type = get<0>(aprilTagInfo);
+        m_targetPose2d = get<1>(aprilTagInfo);
+
+        if (type == DragonDriveTargetFinder::TARGET_INFO::ODOMETRY_BASED)
+        {
+            m_trajectory = CreateDriveToSpeakerTrajectory(currentPose2d, m_targetPose2d);
```

Comment:
instead of needing an if and else if, just make it if (type != DragonDriveTargetFinder::TARGET_FINDER::NOT_FOUND)